cloud/linode/loadbalancers.go

```diff
@@ -296,12 +296,6 @@ func (l *loadbalancers) updateNodeBalancer(
 			return err
 		}
 
-		// Add all of the Nodes to the config
-		var newNBNodes []linodego.NodeBalancerNodeCreateOptions
-		for _, node := range nodes {
-			newNBNodes = append(newNBNodes, l.buildNodeBalancerNodeCreateOptions(node, port.NodePort))
-		}
-
 		// Look for an existing config for this port
 		var currentNBCfg *linodego.NodeBalancerConfig
 		for i := range nbCfgs {
```

Comment:
It's not your code but you can lean on slices here

```
if i := slices.IndexFunc(nbCfgs, func(nbCfg linodego.NodeBalancerConfig) bool {
			return nbCfg.Port == int(port.Port)
		}); i > -1 {
			currentNBCfg = &nbCfgs[i]
		}
```

Reply:
I'll probably not touch this since it is not related to the change we need to do and the PR is already huge enough.